### ISSUE TITLE
False-positive with shorthand object (with `ignoreExportsUsedInFile`)

### ISSUE THREAD

**Issue by onskulit** [webpro/knip#555]
Hello! I have a knip version 4.6.0. But as far as I could make sure, the problem is relevant for the latest version (5.1.2).

I was able to simplify and reproduce it here: https://github.com/onskulit/knip-shorthand-issue/tree/main. There is also a [report](https://github.com/onskulit/knip-shorthand-issue/blob/main/report.txt).

I was able to simplify it to this formulation: knip cannot adequately recognize the object shorthand. When the value of a field is not explicitly set, knip finds this field unused. 

```
export const initialState = { value: 0 };

const counterSlice = createSlice({
  name: 'counter',
  initialState,
})
```

initialState is reported as unused.

When I do this, everything works (there is no errors):
```
export const initialState = { value: 0 };

const counterSlice = createSlice({
  name: 'counter',
  initialState: initialState,
})
```

**Expected behavior**: knip does not report the export if it is used in a file with the `ignoreExportsUsedInFile` flag.

**Actual behavior**: knip reports the export if it is used in the object shorthand.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.1.4. See [Release 5.1.4](https://github.com/webpro/knip/releases/tag/5.1.4) for release notes._

**Comment by webpro** [comment]
Thanks @onskulit, very useful repro too.
